modules/heat_conduction/src/userobjects/SelfShadowSideUserObject.C

```diff
@@ -227,38 +227,26 @@ SelfShadowSideUserObject::check2DIllumination(const Point & qp, const SideIDType
   // loop over all line segments until one is found that provides shade
   for (const auto & line : _lines)
   {
+    const auto & [p1, p2, line_id] = line;
+
     // make sure a side never shades itself
-    if (std::get<2>(line) == id)
+    if (line_id == id)
       continue;
 
-    const bool max0 = std::get<0>(line)(1) > std::get<1>(line)(1);
+    const bool ordered = p1(1) <= p2(1);
 
-    Real y1, y2;
-    if (max0)
-    {
-      y1 = std::get<1>(line)(1);
-      y2 = std::get<0>(line)(1);
-    }
-    else
-    {
-      y1 = std::get<0>(line)(1);
-      y2 = std::get<1>(line)(1);
-    }
+    const auto y1 = ordered ? p1(1) : p2(1);
+    const auto y2 = ordered ? p2(1) : p1(1);
 
     if (y >= y1 && y <= y2)
     {
+      // line segment is oriented parallel to the irradiation direction
+      if (std::abs(y2 - y1) < libMesh::TOLERANCE)
```

Comment:
Usage of this tolerance can potentially be problematic for different sets of units. Maybe something to take into account in the future.

Reply:
yeah, maybe I should put mooseFuzzyequal here. Or actually just do an `== 0` check. I should double check if any valid numerical value other than 0 causes division issues...